server.lisp

```diff
@@ -38,10 +58,13 @@
             (setq *server* socket)
             (real-call))))))
 
-(defvar *remote-host*)
-(defvar *remote-port*)
+(defvar *remote-host* nil
+  "The remote host of a TCP or UDP event. This variable is dynamically bound in the context of a `socket-server', specifically the handler function.")
+(defvar *remote-port* nil
+  "The remote port of a TCP or UDP event. This variable is dynamically bound in the context of a `socket-server', specifically the handler function.")
```

Comment:
To be 100% backwards compatible, you can use 

```lisp
(defvar *remote-host*)
(eval-when (:compile-toplevel :load-toplevel :execute)
  (setf (documentation '*remote-host* 'variable) "xyz..."))
  ```
  
  this way we keep the existing behaviour where the symbol is unbound at toplevel